src/t_stream.c

```diff
@@ -3459,6 +3459,11 @@ void xinfoCommand(client *c) {
 
     /* HELP is special. Handle it ASAP. */
     if (!strcasecmp(c->argv[1]->ptr,"HELP")) {
+        if (c->argc != 2) {
+            addReplySubcommandSyntaxError(c);
+            return;
+        }
+
```

Comment:
shorter, and also consistent with the pattern used in all other HELP sub-commands
```suggestion
    if (c->argc == 2 && !strcasecmp(c->argv[1]->ptr,"HELP")) {
```